openmls/src/treesync/diff.rs

```diff
@@ -290,19 +291,8 @@ impl<'a> TreeSyncDiff<'a> {
         credential_bundle: &CredentialBundle,
     ) -> Result<UpdatePathResult, LibraryError> {
         debug_assert!(self.own_leaf().is_ok(), "Tree diff is missing own leaf");
-        debug_assert!(
```

Comment:
Until we properly fix #1131, the assert should probably stay here. Or am I missing a reason why it has to go now?

Reply:
I removed it because it doesn't look like any of the code that branches from here depends on the existence of `leaf_secret` anymore, but I can definitely throw it back in if it's still a valuable check